templates/bin/setup

```diff
@@ -1,37 +1,37 @@
 #!/usr/bin/env ruby
 require 'pathname'
+require 'fileutils'
+include FileUtils
 
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
 
-def run(name, file_which_must_not_exist = nil)
-  unless file_which_must_not_exist && File.exist?(file_which_must_not_exist)
-    puts "== #{name} =="
-    yield
-    puts
-  end
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
 end
 
-Dir.chdir APP_ROOT do
-  run 'Installing dependencies' do
-    system 'gem install bundler --conservative'
-    system 'bundle check || bundle install --jobs=3 --retry=3'
-  end
+chdir APP_ROOT do
+  # This script is a starting point to setup your application.
+  # Add necessary setup steps to this file.
 
-  run 'Copying config/application.yml', 'config/application.yml' do
-    system 'cp config/application.example.yml config/application.yml'
-  end
+  puts '== Installing dependencies =='
+  system! 'gem install bundler --conservative'
+  system('bundle check') || system!('bundle install')
 
-  run('Preparing database') do
-    system('bin/rails db:version >> /dev/null 2>&1') || system!('bin/rails db:create:all && bin/rails db:setup')
-  end
+  # Install JavaScript dependencies if using Yarn
+  # system('bin/yarn')
 
-  run 'Removing old logs and tempfiles' do
-    system 'rm -f log/*'
-    system 'rm -rf tmp/cache'
+  puts "\n== Copying sample files =="
+  unless File.exist?('config/application.yml')
+    system! 'cp config/application.example.yml config/application.yml'
   end
 
-  run 'Creating pre-push hook link', '.git/hooks/pre-push' do
-    system 'ln -s ../../bin/fastcheck .git/hooks/pre-push'
-  end
+  puts "\n== Preparing database =="
+  system! 'bin/rails db:reset'
```

Comment:
Here there's originally written `bin/rails db:setup`. `db:reset` drops it before, which makes sense in a local environment. Do you agree?

Reply:
doesn't matter. the schema drops the tables anyway.